controllers/tenant/rhoamtenant_controller.go

```diff
@@ -0,0 +1,96 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	usersv1 "github.com/openshift/api/user/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+var log = l.NewLoggerWithContext(l.Fields{l.ControllerLogContext: "tenant_controller"})
+
+// +kubebuilder:rbac:groups=integreatly.org,resources=rhoamtenant,verbs=get;list;watch
+// +kubebuilder:rbac:groups=user.openshift.io,resources=users,verbs=watch;get;list;update
+
+func New(mgr manager.Manager) (*TenantReconciler, error) {
+	restConfig := controllerruntime.GetConfigOrDie()
+	restConfig.Timeout = time.Second * 10
+
+	client, err := k8sclient.New(restConfig, k8sclient.Options{
+		Scheme: mgr.GetScheme(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &TenantReconciler{
+		Client: client,
+		Scheme: mgr.GetScheme(),
+		mgr:    mgr,
+		log:    l.Logger{},
+	}, nil
+}
+
+type TenantReconciler struct {
+	k8sclient.Client
+	Scheme *runtime.Scheme
+	mgr    manager.Manager
+	log    l.Logger
+}
+
+func (r *TenantReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	err := r.addAnnotationToUser(request.Name)
```

Comment:
Can we use request.Name and request.Namespace? Then in addAnnotationToUser we could get the CR, rather than listing all?

Reply:
Updated. Not using listing now, but Get. 
Not using Namespace, as CR is cluster scope